synapse/util/caches/descriptors.py

```diff
@@ -53,9 +53,9 @@
 
 
 class CachedFunction(Generic[F]):
-    invalidate: Any = None
-    invalidate_all: Any = None
-    prefill: Any = None
+    invalidate: Callable[[Tuple[Any, ...]], None]
+    invalidate_all: Callable[[], None]
+    prefill: Callable[[Tuple[Any, ...], Any], None]
```

Comment:
~~Previously these were `Optional` and defaulted to `None`, was that incorrect?~~

Looks like these are always available and never `None`.